### ISSUE TITLE
Transaction tax and UBI

### ISSUE THREAD

**Issue by v1thesource** [kinecosystem/rewards-engine#10]
A previous suggestion of using transaction fees as an anti-spam measure has several implications:

1. Flat transaction fees introduce a price floor for value transfers/interactions. i.e. Any economic activity below $`x`  is disincentivised. Good for spam reduction.

2. % transaction fees are non-regressive (unlike flat fees), but do not establish a price floor. Not so good for spam reduction. 

3. The transaction fees must go somewhere, whether they are burned or rewarded to individuals. **This is the interesting part**.

Points 1 & 2 can be combined to have some minimum flat fee with a small % fee, a la credit card merchant fees. This is a trivial solution for combatting spam while also maintaining proportionality of fees for high-value transactions. The question arises of what to do with the collected fees:

**Burning fees**
Burning some or all of the captured KIN in fees introduces indefinite deflationary pressure. This is bad for stimulating economic activity (which is something we want to facilitate with KIN) because there is an inherent disincentive to spend an appreciating asset. Not to mention a flat fee burn causes supply to tend towards 0 over finite time. This deflationary effect is in addition to that caused by accidental loss of tokens and inactive wallets which cannot be avoided.

**Rewarding miners with fees**
Depending on the consensus mechanism used, all or part of the transaction fees can be used to reward miners for their task of verifying transactions. Most existing blockchains already do this to some extent and it can be argued that this is necessary for ensuring some level of security in the network.

**_Redistribute the fees to users_**
Distribute some of the collected fees to all users in the economy, as a sort of universal basic income (UBI). This provides a continuous stimulus to the economy, encouraging spending and increasing money velocity, especially amongst smaller users with a relatively large marginal propensity to consume. If implemented for Kik users, this will add the unique incentive of remaining on Kik's platform due to the built-in 'loyalty bonus'. The problem of fake accounts being created to take advantage of the UBI may be tackled by implementing a KYC process for those who wish to take part. The participants of the KIN TDE, who have already completed a stringent KYC process, would be an ideal 'test cohort' for the UBI, while also rewarding them for their early contribution to the KIN project.

In summary; consider implementing a combination of flat and % transaction fees on each transaction to reduce spam. Of the transaction fees collected, award part of them to miners, and the remaining part to verified users as a UBI. Avoid a token burn mechanism as this will reduce economic activity.

Discuss.

Edit: The UBI community is widespread, vocal and highly supportive of new technologies. Leverage this bloc.

**Comment by lovelacelibertas** [comment]
Kik users may not be interested in KYC. Most of Kik users want anonymity that's why they use Kik instead of other messaging apps.

UBI distorts incentives in the virtual and in the real world. People/users who do nothing are incentivized. Imagine I receive KIN for having 0 balance in my wallet. I then receive a few KIN from UBI. The first thing I will do is to transfer that KIN into an another empty wallet that is not attached to my identity so I can collect more KIN in the next round of UBI. If people receive KIN for doing nothing then what is the value of this currency? Scarcity brings value to a currency. Users must EARN KIN, not receive it for free. Stimulating an economy by printing currency (i.e. inflation) is a myth.

**Comment by odednoam** [comment]
My understanding is that the consensus among economists is that UBI has serious negative effects on incentives, and only makes sense as an alternative to transfer payments that are subject to inefficient distribution (due to bureaucracy). Plus, in the case of Kin, if you want the redistribution to be uniform you'd have to mandate strong KYC, otherwise you'd be the subject of all kinds of Sybil attacks; most digital services today don't impose strong KYCs and we don't want to interfere with the way they are doing their business -- they know best.

Collecting fees and redistributing them to miners is a good idea and very much like what we're doing. In the case of Kin, the miners are not the people doing the cryptographic verification on the network's integrity, but rather the digital services creating growth in the ecosystem. Currently the KRE plans to use inflationary reserves rather than fees for this purpose, but as commented [here](https://github.com/kinfoundation/kin-rfcs/issues/7), I think a dynamic mixture of both can be used to fuel the rewards engine in such way that actually controls inflation.
